### ISSUE TITLE
java.io.IOException: write failed: EFBIG (File too large)

### ISSUE THREAD

**Issue by hub2git** [TeamNewPipe/NewPipe#4563]
### Checklist

- [x] I am using the latest version. 
- [x] I checked, but didn't find any duplicates of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Actual behaviour

I got an error around the 90% mark

### Expected behavior

I've tried starting the download again. But it just would not continue downloading.
My SD card has a lot of space available. so whatever the problem is, I'm sure the problem is not with running out of space on my SD card.

## Exception
* __User Action:__ download failed
* __Request:__ https://www.youtube.com/watch?v=ClBiOJkDJCc [ {type=video format=MPEG-4 quality=720p videoOnly=false} ]
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ YouTube
* __Version:__ 0.20.1
* __OS:__ Linux samsung/star2qltecs/star2qltecs:10/QP1A.190711.020/G965WVLU7ETG1:user/release-keys 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
java.io.IOException: write failed: EFBIG (File too large)
	at libcore.io.IoBridge.write(IoBridge.java:544)
	at java.io.FileOutputStream.write(FileOutputStream.java:392)
	at us.shandian.giga.io.FileStreamSAF.write(FileStreamSAF.java:131)
	at us.shandian.giga.get.DownloadRunnable.run(DownloadRunnable.java:120)
Caused by: android.system.ErrnoException: write failed: EFBIG (File too large)
	at libcore.io.Linux.writeBytes(Native Method)
	at libcore.io.Linux.write(Linux.java:294)
	at libcore.io.ForwardingOs.write(ForwardingOs.java:241)
	at libcore.io.BlockGuardOs.write(BlockGuardOs.java:416)
	at libcore.io.ForwardingOs.write(ForwardingOs.java:241)
	at libcore.io.IoBridge.write(IoBridge.java:539)
	... 3 more

```
</details>
<hr>

### Device info

 - Android version/Custom ROM version:
 - Device model:

**Comment by TobiGr** [comment]
Is the SD card formatted as internal SD card or is it an external (and thus removable) SD card? If external, is it formatted with FAT32? FAT32 has a file size limit of 4GB

**Comment by hub2git** [comment]
It's an external micro SD card. How can I tell what format the SD card is
in? Is there an app that you can recommend?

**Comment by Stypox** [comment]
I don't know if there is such an app. Did you format the SD on Windows last time, or maybe have you never formatted it?

**Comment by TobiGr** [comment]
Sorry. My bad. I meant whether the external Sd card is formatted as internal _storage_ or just as external storage. If it is just external storage, you can remove it and plug into a SD card slot of your computer (after putting it into an adapter). Going to the drives properties should show its file system.
I really expect this to be a FAT32 file system, because the only two remaing by Android supported file systems are ext3 and ext4. Both  can handle larger files, but are not compatible with Windows (at least not by default).

**Comment by hub2git** [comment]
I ran a few apps (
https://play.google.com/store/apps/details?id=ru.andr7e.deviceinfohw&hl=en_CA&gl=US;
https://play.google.com/store/apps/details?id=me.kuder.diskinfo), and both
say that the microSD card's filesystem is sdcardfs.

My computer runs Linux Mint, so I guess there's no need to have a FAT32
microSD card.

**Comment by TobiGr** [comment]
Thanks. Files on sdcardfs are also limited to 4GB. We should check that before starting the download.

A possible solution to the problem would be to split the the files. However, that' requires some work.

**Comment by opusforlife2** [comment]
Or just add an alternate location in the internal storage, and offer to save the file there if it is over 4 GB.

If we don't want to add another setting for this, we could just offer a dialogue box saying something like "Files over 4 GB cannot be saved to the SD Card. Save to Internal Storage/Newpipe/ instead? Yes/No".
